src/components/createConnectDecorator.js

```diff
@@ -1,24 +1,121 @@
+import createStoreShape from '../utils/createStoreShape';
 import getDisplayName from '../utils/getDisplayName';
 import shallowEqualScalar from '../utils/shallowEqualScalar';
+import shallowEqual from '../utils/shallowEqual';
+import isPlainObject from '../utils/isPlainObject';
+import wrapActionCreators from '../utils/wrapActionCreators';
+import invariant from 'invariant';
 
-export default function createConnectDecorator(React, Connector) {
-  const { Component } = React;
+const emptySelector = () => ({});
 
-  return function connect(select) {
-    return DecoratedComponent => class ConnectorDecorator extends Component {
-      static displayName = `Connector(${getDisplayName(DecoratedComponent)})`;
+const emptyBinder = () => ({});
+
+const identityMerge = (slice, actionsCreators, props) => ({...slice, ...actionsCreators, ...props});
```

Comment:
We ran into some annoying bugs with a merge like this, specifically the fact that props can override slice. Given that Connect's primary responsibility is getting state from the store and passing it along, I'm not sure if allowing props to override it by default is right.

Some other options:

1. Put props first, allowing slice and actionCreators to override it.
2. Do not merge props in by default, a custom merge can be passed in to do this explicitly if desired. 

For what it's worth, the specific problem we ran into involved nested connectors and the first connector splatting its props to the 2nd. They both had a `loaded` flag, but they meant different things. Nested connectors can happen especially with nested routes.